File: Anul II/Semestrul I/DAW/SissaCoffee/backend/SissaCoffee/Models/Product.cs
using SissaCoffee.Models.Base;

namespace SissaCoffee.Models;

public class Product: BaseEntity
{
    public string Name { get; set; } = String.Empty;

    public Tag? Tag { get; set; }
    public Guid? TagId { get; set; }

    public ProductVariant Variant { get; set; } = new();
    public Guid VariantId { get; set; } = Guid.Empty;

    public ICollection<ProductIngredient> Ingredients { get; set; } = new List<ProductIngredient>();
}

File: Anul II/Semestrul I/DAW/SissaCoffee/frontend/src/app/interfaces/product.ts
import {IProductVariant} from "./product-variant";

export interface IProduct {
  id?: string;
  name: string;
  tag?: string;
  variant: IProductVariant;
  ingredients?: string[];
}

export interface IProductCreate {
  name: string;
  variant: string;
  ingredients: string[];
  tag?: string;
}
